{{cookiecutter.project_slug}}/docker/entrypoint.sh

```diff
@@ -11,43 +11,67 @@ echo "  User    :`id $(whoami)`"
 echo "  Workdir :`pwd`"
 
 
-if [[ ${SC_BUILD_TARGET} == "development" ]]
-then
-    # NOTE: expects docker run ... -v $(pwd):/devel/services/{{ cookiecutter.project_slug }}
-    DEVEL_MOUNT=/devel/services/{{ cookiecutter.project_slug }}
+# expect input/output/log folders to be mounted
+stat $INPUT_FOLDER &> /dev/null || \
+        (echo "ERROR: You must mount '$INPUT_FOLDER' to deduce user and group ids" && exit 1)
+stat $OUTPUT_FOLDER &> /dev/null || \
+    (echo "ERROR: You must mount '$OUTPUT_FOLDER' to deduce user and group ids" && exit 1)
+stat $LOG_FOLDER &> /dev/null || \
+    (echo "ERROR: You must mount '$LOG_FOLDER' to deduce user and group ids" && exit 1)
 
-    stat $DEVEL_MOUNT &> /dev/null || \
-        (echo "ERROR: You must mount '$DEVEL_MOUNT' to deduce user and group ids" && exit 1) # FIXME: exit does not stop script
+stat $INPUT_FOLDER &> /dev/null
+if [[ $? -eq 0 ]]
+then
+    # NOTE: expects docker run ... -v /path/to/input/folder:$INPUT_FOLDER
+    # check input/output/log folders are owned by the same user
+    if [[ $(stat -c %u $INPUT_FOLDER) -ne $(stat -c %u $OUTPUT_FOLDER) ]]
+    then
+        (echo "ERROR: '$INPUT_FOLDER' and '$OUTPUT_FOLDER' have different user id's. not allowed" && exit 1)
+    elif [[ $(stat -c %u $INPUT_FOLDER) -ne $(stat -c %u $LOG_FOLDER) ]]
+    then
+        (echo "ERROR: '$INPUT_FOLDER' and '$LOG_FOLDER' have different user id's. not allowed" && exit 1)
+    fi
+    # check input/output/log folders are owned by the same group
+    if [[ $(stat -c %g $INPUT_FOLDER) -ne $(stat -c %g $OUTPUT_FOLDER) ]]
+    then
+        (echo "ERROR: '$INPUT_FOLDER' and '$OUTPUT_FOLDER' have different group id's. not allowed" && exit 1)
+    elif [[ $(stat -c %g $INPUT_FOLDER) -ne $(stat -c %g $LOG_FOLDER) ]]
+    then
+        (echo "ERROR: '$INPUT_FOLDER' and '$LOG_FOLDER' have different group id's. not allowed" && exit 1)
+    fi
 
-    USERID=$(stat -c %u $DEVEL_MOUNT)
-    GROUPID=$(stat -c %g $DEVEL_MOUNT)
+    USERID=$(stat -c %u $INPUT_FOLDER)
+    GROUPID=$(stat -c %g $INPUT_FOLDER)
     GROUPNAME=$(getent group ${GROUPID} | cut -d: -f1)
-
     if [[ $USERID -eq 0 ]]
     then
         addgroup scu root
```

Comment:
I would parametrize scu as well. Perhaps we should add ``SC_USER_NAME`` variable in ``Dockerfile``.

Reply:
done.